server/channels/app/platform/web_hub.go

```diff
@@ -619,30 +630,55 @@ func areAllInactive(conns []*WebConn) bool {
 // It requires 3 functionalities which need to be very fast:
 // - check if a connection exists or not.
 // - get all connections for a given userID.
+// - get all connections for a given channelID.
 // - get all connections.
 type hubConnectionIndex struct {
 	// byUserId stores the list of connections for a given userID
 	byUserId map[string][]*WebConn
+	// byChannelID stores the list of connections for a given channelID.
+	byChannelID map[string][]*WebConn
 	// byConnection serves the dual purpose of storing the index of the webconn
 	// in the value of byUserId map, and also to get all connections.
 	byConnection   map[*WebConn]int
 	byConnectionId map[string]*WebConn
 	// staleThreshold is the limit beyond which inactive connections
 	// will be deleted.
 	staleThreshold time.Duration
+
+	store  store.Store
+	logger mlog.LoggerIFace
 }
 
-func newHubConnectionIndex(interval time.Duration) *hubConnectionIndex {
+func newHubConnectionIndex(interval time.Duration,
+	store store.Store,
+	logger mlog.LoggerIFace,
+) *hubConnectionIndex {
 	return &hubConnectionIndex{
 		byUserId:       make(map[string][]*WebConn),
+		byChannelID:    make(map[string][]*WebConn),
 		byConnection:   make(map[*WebConn]int),
 		byConnectionId: make(map[string]*WebConn),
 		staleThreshold: interval,
+		store:          store,
+		logger:         logger,
 	}
 }
 
 func (i *hubConnectionIndex) Add(wc *WebConn) {
 	i.byUserId[wc.UserId] = append(i.byUserId[wc.UserId], wc)
+
+	cm, err := i.store.Channel().GetAllChannelMembersForUser(request.EmptyContext(i.logger), wc.UserId, false, false)
+	if err != nil {
+		// In this case, the map won't be populated. This is effectively the same
+		// as not broadcasting the event, and that's how it was implemented in
+		// platform/web_conn.go
+		i.logger.Warn("Error while getting GetAllChannelMembersForUser", mlog.Err(err))
```

Comment:
I am still not happy with this. Thinking more, if this failed inside web_conn.go, only one connection gets affected, but if this fails here, then a large portion of users will be affected. We cannot continue with a failure. Will improve this.